mono/mini/intrinsics.c

```diff
@@ -755,6 +756,15 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_PADD_IMM, dreg, args [0]->dreg, MONO_STRUCT_OFFSET (MonoArray, vector));
 			return ins;
 		}
+		else if (!strcmp (cmethod->name, "GetElementSize")) {
+			int vt_reg = alloc_preg (cfg);
+			int class_reg = alloc_preg (cfg);
+			int sizes_reg = alloc_preg (cfg);
```

Comment:
`sizes_reg` needs to be `alloc_ireg`? also I guess you can re-use `vt_reg` for the second load (`vt_reg, vt_reg`).

Reply:
You're right. It ends up implemented using the same code but I should be using the right one. Reusing the register would probably work but not sure how beneficial it is. For LLVM we convert to SSA anyway.

Reply:
yeah I know, it's just a nit - less code